training/src/anemoi/training/train/train.py

```diff
@@ -167,6 +167,66 @@ def model(self) -> GraphForecaster:
 
         model = GraphForecaster(**kwargs)
 
+        # Apply checkpoint loading and model modifications
+        return self._apply_checkpoint_pipeline(model, **kwargs)
+
+    def _apply_checkpoint_pipeline(self, model: GraphForecaster, **kwargs) -> GraphForecaster:
+        """Apply checkpoint loading and model modifications using the pipeline system.
+
+        This method provides backward compatibility with legacy configuration while
+        using the modern checkpoint pipeline system internally.
+
+        Parameters
+        ----------
+        model : GraphForecaster
+            The model to apply checkpoint loading to
+        **kwargs : dict
+            Additional keyword arguments for GraphForecaster
+
+        Returns
+        -------
+        GraphForecaster
+            Model with checkpoint loaded and modifications applied
+        """
+        # Check if we should use the modern checkpoint pipeline
+        if hasattr(self.config.training, "checkpoint_pipeline") and self.config.training.checkpoint_pipeline:
+            return self._use_modern_checkpoint_pipeline(model)
+
+        # Use legacy checkpoint logic for backward compatibility
+        return self._use_legacy_checkpoint_logic(model, **kwargs)
+
+    def _use_modern_checkpoint_pipeline(self, model: GraphForecaster) -> GraphForecaster:
+        """Use the modern checkpoint pipeline system."""
+        try:
+            import asyncio
```

Comment:
is the asyncio functionality optional? (as in could people use the pipeline without this?)